### PR TITLE
Add table for FreeBSD kernel modules.

### PR DESCRIPTION
**Pull request by lxcode**
Monitor kernel modules using kldstat(2) and friends.

### CODE REVIEW

File: osquery/tables/system/freebsd/kernel_modules.cpp [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <fstream>
+#include <iomanip>
+#include <sys/param.h>
```
Comment (jedi22) [inline osquery/tables/system/freebsd/kernel_modules.cpp]: nit, newline

File: osquery/tables/system/freebsd/kernel_modules.cpp [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+QueryData genKernelModules(QueryContext& context) {
+  QueryData results;
+
+  int fileid;
+  for (fileid = kldnext(0); fileid > 0; fileid = kldnext(fileid)) {
+    Row r;
+    struct kld_file_stat stat;
+    stat.version = sizeof(struct kld_file_stat);
+    if (kldstat(fileid, &stat) < 0) {
+      VLOG(1) << "Cannot stat module";
```
Comment (jedi22) [inline osquery/tables/system/freebsd/kernel_modules.cpp]: Can you use LOG(INFO)?

File: osquery/tables/system/freebsd/kernel_modules.cpp [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+      return {};
```
Comment (jedi22) [inline osquery/tables/system/freebsd/kernel_modules.cpp]: Do you want to bail on returning anything when you error? or just continue on to the next one?

**Comment by theopolis** [comment]
test freebsd please

File: osquery/tables/system/freebsd/kernel_modules.cpp (hunk 3 above)
Comment (lxcode) [inline osquery/tables/system/freebsd/kernel_modules.cpp]: I can't think of a situation where one module successfully stats and another one doesn't — it probably just means it's all broken, so I think this is ok.

File: osquery/tables/system/freebsd/kernel_modules.cpp (hunk 2 above)
Comment (lxcode) [inline osquery/tables/system/freebsd/kernel_modules.cpp]: Sure, but would maybe LOG(ERROR) be more appropriate?

**Comment by jedi22** [comment]
ok to test

**Comment by osqueryer** [comment]
:no_good: The commit d624e9d523bcab03ea4f3066fe36a89b671c4013 (Job results: [1673](https://jenkins.osquery.io/job/osqueryPullRequestAudit/1673/)) failed the code audit or documentation test.

**Comment by osqueryer** [comment]
:-1: The commit d624e9d523bcab03ea4f3066fe36a89b671c4013 (Job results: [999](https://jenkins.osquery.io/job/osqueryPullRequestOSX/999/)) failed one or more tests (**macOS/OS X**).

**Comment by osqueryer** [comment]
:-1: The commit d624e9d523bcab03ea4f3066fe36a89b671c4013 (Job results: [1000](https://jenkins.osquery.io/job/osqueryPullRequestOSX/1000/)) failed one or more tests (**macOS/OS X**).

**Comment by osqueryer** [comment]
:-1: The commit d624e9d523bcab03ea4f3066fe36a89b671c4013 (Job results: [1737](https://jenkins.osquery.io/job/osqueryPullRequestWindows/1737/)) failed one or more tests (**Windows**).

**Comment by osqueryer** [comment]
:-1: The commit d624e9d523bcab03ea4f3066fe36a89b671c4013 (Job results: [1738](https://jenkins.osquery.io/job/osqueryPullRequestWindows/1738/)) failed one or more tests (**Windows**).

**Comment by lxcode** [comment]
clang-format's suggestion puts the includes in the wrong order and breaks the build. 

macOS breakage appears unrelated to this diff.

**Comment by poppyseedplehzr** [comment]
You can prevent clang-format from getting upset about import errors with:
```
// clang-format off
...
// clang-format on
```

**Comment by osqueryer** [comment]
:-1: The commit afc7ad2852ffbaddb06e56b9fcf8db736144d57a (Job results: [1003](https://jenkins.osquery.io/job/osqueryPullRequestOSX/1003/)) failed one or more tests (**macOS/OS X**).

**Comment by osqueryer** [comment]
:-1: The commit afc7ad2852ffbaddb06e56b9fcf8db736144d57a (Job results: [1741](https://jenkins.osquery.io/job/osqueryPullRequestWindows/1741/)) failed one or more tests (**Windows**).

**Comment by osqueryer** [comment]
:-1: The commit afc7ad2852ffbaddb06e56b9fcf8db736144d57a (Job results: [1004](https://jenkins.osquery.io/job/osqueryPullRequestOSX/1004/)) failed one or more tests (**macOS/OS X**).

**Comment by osqueryer** [comment]
:-1: The commit afc7ad2852ffbaddb06e56b9fcf8db736144d57a (Job results: [1742](https://jenkins.osquery.io/job/osqueryPullRequestWindows/1742/)) failed one or more tests (**Windows**).

**Pull request merged**